### ISSUE TITLE
Ability to pass Spec as Object does not appear to work for Yaml files

### ISSUE THREAD

**Issue by GSDTN** [RuntimeTools/OpenAPIValidators#48]
**OpenAPI version**
3

**Describe the bug**
I have a spec that works if I pull it down, write it to a file and then pass that file path into chaiResponseValidator(specPath).
After pulling down 0.7.0 - I was expecting that the feature in 0.6.0 to just do an HTTP get and pass the data portion of the response in would work.
However, it appears that the code is inspecting  the input and looking for an Object for that capability and if it sees a string, it assumes it is a path.
The Object approach possibly works with JSON files, but does not work with YAML.
I get the following error: The provided argument must be either an absolute filepath or an object representing an OpenAPI specification.  Followed by a print out of the YAML spec.
I recommend you inspect the string for line feeds for 

**To Reproduce**
1. Pull down a YAML API Spec using: const apiSpec = (await axios.get("https://content-services.dtn.com/api-specs/weather-api/weather-oas-1.3.1.yaml", {"headers": specHeaders})).data;
2. Pass this in: chai.use(chaiResponseValidator(apiSpec));
3. See error

**Comment by rwalle61** [comment]
Thanks for raising this and investigating the cause. I think you're right - if it sees a string input, it assumes it is a path. I did not anticipate that when axios fetches a yaml file, `response.data` could be a string rather than an object.

Presumably if you parse the yaml string as an object (e.g.g 
```js
chai.use(chaiResponseValidator(yaml.safeLoad(string)));
```
Then this works?

If so, then it should be fairly simple for me/you/someone to integrate that logic into our plugin, so that in future you don't have to do it manually

**Comment by GSDTN** [comment]
Using yaml.safeLoad(spec) works.

I may have some Windows based character line feed issues so I have the following you could add:
Or just let me know if you would like me to commit into a new branch.

openApiSpecFactory.js
…
function loadSpec(arg) {
  const argType = typeOf(arg);
  try {
    if (argType === 'string') {
      //If the length of the string is less than 260 characters (the maximum path length permitted in Windows)
      //Simplifies the checking for paths across various platforms
      if (arg.length < 260) {
        return loadFile(arg);
      } else {
        try {
          return yaml.safeLoad(arg);
        } catch (error) {
          throw new Error(`Invalid YAML or JSON:\n${error.message}`);
        }
      }
    } else if (argType === 'object') {
      return arg;
    } else {
      throw new Error(`Received type '${argType}'`);
    }
  } catch (error) {
    throw new Error(`The provided argument must be either an absolute filepath or an object representing an OpenAPI specification.\nError details: ${error.message}`);
  }
}


setup.test.js
…
const fs = require('fs-extra');
const filefs = require('fs');  //New import to read file
const chaiResponseValidator = require('../..');
…
//New tests around line 120

describe('Yaml string representing a valid OpenAPI file', function () {
  it('returns a function', function () {
    const pathToApiSpec = path.resolve('test/resources/exampleOpenApiFiles/valid/openapi3.yml');
    const apiSpec = filefs.readFileSync(pathToApiSpec, { encoding: "UTF8" });
    expect(chaiResponseValidator(apiSpec)).to.be.a('function');
  });
});
describe('JSON string representing a valid OpenAPI file', function () {
  it('returns a function', function () {
    const pathToApiSpec = path.resolve('test/resources/exampleOpenApiFiles/valid/openapi3.json');
    const apiSpec = filefs.readFileSync(pathToApiSpec, { encoding: "UTF8" });
    expect(chaiResponseValidator(apiSpec)).to.be.a('function');
  });
});

**Comment by rwalle61** [comment]
Cool, yea I'm very happy to review a PR and merge, if you'd like to raise one. 

The tests are already at 100% coverage and mutation score (do `npm run test:full`) so it'd be great if you can maintain that, should be straightforward as you're only adding a few lines

Also just to confirm - I assume because you added this test
```
const pathToApiSpec = path.resolve('test/resources/exampleOpenApiFiles/valid/openapi3.json');
    const apiSpec = filefs.readFileSync(pathToApiSpec, { encoding: "UTF8" });
```
that axios returns yaml files as that kind of string? Maybe clarify that in a variable name or comment

**Comment by rwalle61** [comment]
@GSDTN is this still an issue? If so I will look at resolving it soon

**Comment by rwalle61** [comment]
On reflection, I think this is not a bug, because our README and error message already explain that the provided argument must be either 
1. a string filepath or 
2. an object representing a valid OpenAPI spec.

So this issue is suggesting we enhance our setup, to also support:

3. a stringified object representing a valid OpenAPI spec

But 3 is a mix of 1 and 2, and I'm not sure we should take on that complexity. I think it's clearest for everyone if the user explicitly gives us 1 or 2, without us guessing which it is (which may lead to problems down the line).

tl;dr, do 
```js
chai.use(chaiResponseValidator(yaml.safeLoad(string));
```
or 
```js
jestOpenAPI(yaml.safeLoad(string));
```

If users disagree, feel free to reopen this, or raise a PR :)
